src/core/grid_based_algorithms/lb_particle_coupling.cpp

```diff
@@ -283,20 +248,25 @@ void lb_lbcoupling_calc_particle_lattice_ia(
         };
 
         auto couple_particle = [&](Particle &p) -> void {
+          // We only couple ghosts, if the physical particle is not on the node
+          if (p.l.ghost and not local_particles[p.p.identity]->l.ghost)
```

Comment:
Are you sure this is correct? At least with the old scheme this would be wrong...

Reply:
I think this has to be handled by Walberla, sine it depends on how many ghost layers are available and what directions are periodic. Ingos Pr against the Walberla branch should sort that out. The lb_momentum_conservation test will tell us, whether or not we got it right.

Reply:
Ok, but this would mean that there is a halo reduction (of the force density) anyway?